17548: Personal details push to page

gedu opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->
Updated dropdowns fields to be MenuItems which sends the user to another page to pick the option
Standarized the spaces between Inputs, created a new FormSpace component.

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/17548
PROPOSAL: https://github.com/Expensify/App/issues/17548#issuecomment-1521510794


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

Errors on Empty State
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Save
4. All the Inputs even the Menu Items should show a required field error

Update the menu Country item (no USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select a new country (no USA)
5. The Country Item should show the selected Country
6. The State Input should be an input component
7. You can select new ones and the Menu Item should be updated

Update the menu Country item (USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select United State
5. Should send you back and the Country Item should show the selected Country
6. The State Input should be a Menu Item

Update the menu State item (USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select United State
5. The Country Item should show the selected Country
6. Clicking on the State menu should send you to a new Page
7. You can select any State
8. Should send you back to and show you the updated State

Autocomplete address
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Start filling the Address Line 1, a suggestion menu should shows up
4. Clicking on any address should autocomplete, menu items and inputs

- [ ] Verify that no errors appear in the JS console

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

Errors on Empty State
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Save
4. All the Inputs even the Menu Items should show a required field error

Update the menu Country item (no USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select a new country (no USA)
5. The Country Item should show the selected Country
6. The State Input should be an input component
7. You can select new ones and the Menu Item should be updated

Update the menu Country item (USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select United State
5. Should send you back and the Country Item should show the selected Country
6. The State Input should be a Menu Item

Update the menu State item (USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select United State
5. The Country Item should show the selected Country
6. Clicking on the State menu should send you to a new Page
7. You can select any State
8. Should send you back to and show you the updated State

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
10. Upload an image via copy paste
11. Verify a modal appears displaying a preview of that image
--->

Errors on Empty State
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Save
4. All the Inputs even the Menu Items should show a required field error

Update the menu Country item (no USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select a new country (no USA)
5. The Country Item should show the selected Country
6. The State Input should be an input component
7. You can select new ones and the Menu Item should be updated

Update the menu Country item (USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select United State
5. Should send you back and the Country Item should show the selected Country
6. The State Input should be a Menu Item

Update the menu State item (USA)
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Click on Country item
4. Should send you to a new page, select United State
5. The Country Item should show the selected Country
6. Clicking on the State menu should send you to a new Page
7. You can select any State
8. Should send you back to and show you the updated State

Autocomplete address
1. Sign in using an User that doesn't have address
2. Go to the Address Page (No inputs should be filled)
3. Start filling the Address Line 1, a suggestion menu should shows up
4. Clicking on any address should autocomplete, menu items and inputs

- [ ] Verify you can update the Menu Items or you can go back to from the selection and the last selection should be in place
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/1676818/236286530-2cba05a5-b1f3-4550-8eb6-7bfeb1a6ccad.mp4


https://user-images.githubusercontent.com/1676818/236286664-d3535992-d042-4ac8-95df-d86fd00a50c3.mp4


https://user-images.githubusercontent.com/1676818/236287012-a34543bc-5eef-4a3e-84eb-e729cbca918f.mp4




</details>

<details>
<summary>Mobile Web - Chrome</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/1676818/236286832-11326a15-dddb-4913-8b1e-9039649358bd.mp4




</details>

<details>
<summary>Mobile Web - Safari</summary>

<!-- add screenshots or videos here -->


https://user-images.githubusercontent.com/1676818/236286951-f0fa8418-21f1-405d-973b-d52a5b5d3eab.mp4



</details>

<details>
<summary>Desktop</summary>

<!-- add screenshots or videos here -->

https://user-images.githubusercontent.com/1676818/236287076-1aeeb693-63da-4983-9610-0fff50b804bb.mp4




</details>

<details>
<summary>iOS</summary>

<!-- add screenshots or videos here -->


https://user-images.githubusercontent.com/1676818/236287166-05ae59ac-8df6-4454-b64b-3373c3fb22ce.mp4



</details>

<details>
<summary>Android</summary>

<!-- add screenshots or videos here -->


https://user-images.githubusercontent.com/1676818/236287217-78af4814-55cf-4051-8eea-e16f4d346019.mp4



</details>

best-lucky1030 reviewed (commented):
seems like mentioned issues were fixed and on my local side, it works. 👍

ArekChr commented on src/pages/CountrySelectorPage.js:
```diff
@@ -0,0 +1,98 @@
+    return _.filter(countries, country => country.text.toLowerCase().includes(searchValue.toLowerCase()));
+}
+
+function CountrySelectorPage(props) {
```
DRY, This file has a lot of similarities with the `StateSelectorPage` component. Could you consider refactoring this code to create more reusable components that can be shared across multiple files?

ArekChr commented on src/libs/actions/PersonalDetails.js:
```diff
@@ -96,6 +96,14 @@ function getCountryISO(countryName) {
     return _.findKey(CONST.ALL_COUNTRIES, country => country === countryName) || '';
 }
 
+function getCountryNameBy(countryISO) {
```
```suggestion
/**
 * Returns the name of the country associated with the provided ISO code.
 * If the provided code is invalid, an empty string is returned.
 * 
 * @param {string} countryISO The ISO code of the country to look up.
 * @returns {string} The name of the country associated with the provided ISO code.
 */
function getCountryNameBy(countryISO) {
```

ArekChr commented on src/pages/settings/Profile/PersonalDetails/AddressPage.js:
```diff
@@ -52,23 +52,41 @@ const defaultProps = {
     },
 };
 
+function FormSpace() {
```
Consider encapsulating FormSpace in the AddressPage, as it is only used in one component.
Or just use <View style={styles.formSpaceVertical} /> instead.

ArekChr commented on src/components/StatePicker.js:
```diff
@@ -1,67 +1,102 @@
+        Navigation.navigate(ROUTES.getUsaStateSelectionRoute(stateTitle || stateISO, Navigation.getActiveRoute()));
+    }, [stateTitle, stateISO]);
+
+    const title = useMemo(() => {
```
I think memoization is not needed here

ArekChr replied:
I propose to write a separate function for that

gedu replied:
If I move it into a function I will need to pass the props, and according to the guidelines I will wrap it into a `useCallback` so will look kind the same but will recalculate the same values so I would keep it into a memo given that other components can cause re-renders

<img width="185" alt="Screen Shot 2023-05-05 at 16 55 33" src="https://user-images.githubusercontent.com/1676818/236493668-9269ed6a-1e0c-4131-8a6f-4d65f8d51c76.png">

ArekChr replied:
But this is a string type, which shouldn't cause a re-render in case the title stays the same without using useMemo, that because I requested removing that because it's a cheap function

ArekChr commented on src/components/StatePicker.js:
```diff
@@ -1,67 +1,102 @@
+    const stateISO = props.stateISO;
+    const [stateTitle, setStateTitle] = useState(stateISO);
+    const paramStateISO = lodashGet(route, 'params.stateISO');
+    const navigation = props.navigation;
+    const onInputChange = props.onInputChange;
+    const defaultValue = props.defaultValue;
+    const translate = props.translate;
```
Too many unnecessary declarations here, to improve readability please destruct props in component or call functions with just a simple approach `props.navigation.navigate(...)`

gedu replied:
Yeah, I talked internally about this but seems that we aren't gonna do [destruction](https://expensify.slack.com/archives/C01GTK53T8Q/p1680718704040129), but probably we can bring it back in discussion

ArekChr replied:
This topic is mainly about `defaultProps` with destructing, but let's leave destructing until that is clarified.
I propose to use simple syntax `props.navigation.navigate` to reduce variable declaration, which will improve the hook's readability right now, it looks a bit messy

gedu commented on src/pages/settings/Profile/PersonalDetails/AddressPage.js:
```diff
@@ -164,12 +188,14 @@ class AddressPage extends Component {
                     enabledWhenOffline
                 >
-                    <View style={styles.mb4}>
+                    <View>
```
I don't have it in my code, checking

gedu commented on src/pages/settings/Profile/PersonalDetails/AddressPage.js:
```diff
@@ -52,23 +52,41 @@ const defaultProps = {
         const zipSampleFormat = lodashGet(CONST.COUNTRY_ZIP_REGEX_DATA, [currentCountryISO, 'samples'], '');
+
         this.state = {
             isUsaForm: (currentCountry === CONST.COUNTRY.US || currentCountry === CONST.USA_COUNTRY_NAME),
```
it is also being used in validate, and validate get executed many times and in parallel from other components, I wouldn't move it out

ArekChr replied:
Agree. I see now. It's used in validating function

gedu commented on src/pages/StateSelectorPage.js:
```diff
@@ -0,0 +1,115 @@
+        customIcon: currentCountryState === state.stateISO ? greenCheckmark : undefined,
+    })), [translate, currentCountryState]);
+
+    const updateCountryState = useCallback((selectedState) => {
```
They prefer to use `useCallback` you can read the new [guidelines here](https://github.com/Expensify/App/pull/16882/files)

ArekChr replied:
But in this case, it doesn't help at all. It's useful when a function is passed to PureComponent or memoized component or if there are computationally expensive calculations.
But here it is not. SelectionList doesn't implement memorization. UpdateCountryState is a cheap function, so memoizing this function consumes unnecessary js memory IMO

gedu commented on src/pages/StateSelectorPage.js:
```diff
@@ -0,0 +1,115 @@
+        Navigation.navigate(`${route.params.backTo}?stateISO=${selectedState.value}`);
+    }, [route]);
+
+    const filteredCountryStates = filterCountryStates(searchValue, countryStates);
```
Maybe doesn't make sense to memoize because `searchValue` will change on every user input, so `useMemo` will change everytime

gedu replied:
`countryStates` worth memo, because on every key press, the component will get re-render, this means if the user wanna type "United State", when the User type "U" will cause a re-render, same when type "n"... I don't wanna re-create the `countryStates` because they will be the same always. Memo `filterCountryStates` in this case don't worth it because it will depend on `searchValue` which will change on every key press, so the filter will recalculated. 
Not all the component depends on `searchValue`

gedu commented on src/pages/StateSelectorPage.js:
```diff
@@ -0,0 +1,115 @@
+    const route = props.route;
+    const currentCountryState = route.params.stateISO || lodashGet(props.privatePersonalDetails, 'address.state');
+
+    const countryStates = useMemo(() => _.map(translate('allStates'), state => ({
```
Explained above maybe we shouldn't memo `filteredCountryStates`

best-lucky1030 reviewed (commented):
we have to optimize the codebase.

ArekChr commented:
except for minor corrections that the guidelines are in the process of clarifying, overall, LGTM 👌

gedu commented:
@best-lucky1030 No, is ready for Expensify review

best-lucky1030 reviewed (commented):
@gedu seems like we are still getting lint error.

gedu commented:
@venture1981 all solved

cristipaval commented:
bump @mananjadhav

mananjadhav commented:
Taking longer than expected, reviewed half the files, reviewing the other half and will test it as well.

venture1981 reviewed (commented):
done!

mananjadhav commented:
I reviewed and I am fine with the changes, but I see an outstanding comment on making it DRY - StateSelector and CountrySelector. Are we looking to fix this right now ? or follow up in another PR.

cristipaval commented:
I just thought now that it would be a great if @shawnborton has some time to review this from the design standpoint 🙏

0xmiroslav reviewed (commented):
`OptionsSelector` already supports search. Why can't we reuse that component?
`PronounsPage` is a good example. I don't see major UI difference between country/state selector pages and pronouns page.
Also let's implement what pronouns page does:
- pre-populate search input and auto-scroll to selected item
- deselect support
- trim search

https://github.com/Expensify/App/assets/97473779/9820cdac-28b7-4c27-84b9-b4226c18f8aa

shawnborton commented:
> Hover background area is not familiar to me. Is this design approved? I'd expect to cover full width and enough vertical padding like other menu items.

I agree, that does not look great. I agree that the hover styles should match full width like places in the app.

shawnborton commented:
For these push-to-page rows, are we using a global component? I was surprised to see that the label font size is smaller when there is no value:
<img width="268" alt="image" src="https://github.com/Expensify/App/assets/2319350/91e2b694-a3d9-45dd-85f0-f76ed50fea24">


For that "emtpy state" of the push row when there is no value set, the font size should be 15px. Can we make sure that is the case for all of these rows, everywhere? cc @cristipaval

Then once a value is set, the label size goes to 13px and it rests above the value, which is at 15px.

shawnborton commented:
For the list selection views, @cristipaval where do we stand with updating the styles there? we shouldn't have the lines go edge-to-edge, and I thought we had an issue open to make sure the list selection is consistent everywhere.

<img width="267" alt="image" src="https://github.com/Expensify/App/assets/2319350/aea162ce-5c29-4d03-ad57-0e4af785435e">

shawnborton commented:
It feels weird to me that state comes before City here:
<img width="264" alt="image" src="https://github.com/Expensify/App/assets/2319350/3ec80652-c542-49bc-89e5-f91f2338450a">

gedu commented:
Hey, I will review the next bugs:
- Hover style making it full width 
- Will take a look at `shouldSaveDraft`
- State should be Name and not code

The rest is already happening probably will be good to take it into another issue?

> Why can't we reuse that component?

I did it like that the first time but got the review to Don't Repeat Yourself, so I made them use one component

cristipaval commented:
> For the list selection views, @cristipaval where do we stand with updating the styles there? we shouldn't have the lines go edge-to-edge, and I thought we had an issue open to make sure the list selection is consistent everywhere.
> 
> <img alt="image" width="267" src="https://user-images.githubusercontent.com/2319350/241766034-aea162ce-5c29-4d03-ad57-0e4af785435e.png">

[Here](https://github.com/Expensify/App/issues/11795#issuecomment-1543877225) is the plan to do the massive refactor. So it might take a while for this refactor to be finished. Do you think we should fix the lists which have lists that go edge to edge before the refactoring? We could do this if you think that this can't be held by the refactor.

shawnborton commented:
Ah, so just to make sure I understand, the list style in this PR already existed and wasn't implemented as part of this PR? If so, that's probably fine and we can wait until the refactor to fix it.

cristipaval commented:
>  the list style in this PR already existed and wasn't implemented as part of this PR?

Ah, you're right! The picker screen was implemented as part of this PR. Even if the component for the list item is not yet refactored, we should make it look as per the requirement.

Thanks a lot for the review @shawnborton 🙇 

@gedu Could you please address the @shawnborton feedback?

gedu commented:
Yes, working on it today

gedu commented:
@cristipaval pushed the fixes listed


https://github.com/Expensify/App/assets/1676818/d9a55faa-0070-4f94-8a09-66e329b20429


https://github.com/Expensify/App/assets/1676818/c0bd84ab-4116-42c9-83d7-d026b5d61bff

Note: didn't wanna add too much vertical padding to the menu items because when not hovered seems that there is a lot of space between items

shawnborton commented:
For this here:
<img width="256" alt="image" src="https://github.com/Expensify/App/assets/2319350/8750567e-5327-421f-96aa-536f9aad4e6d">


It feels like the label is too far away from the value. And it doesn't quite look like there is enough padding on the top/bottom. Can you make sure this looks identical to the profile page here: 
<img width="385" alt="image" src="https://github.com/Expensify/App/assets/2319350/f13a3e3c-21d8-467f-b035-78d7188cbdae">


Any reason why these aren't all using the same component either? I feel like we keep rewriting this same pattern.

shawnborton commented:
For list views, we want the style to look like this:
![image](https://github.com/Expensify/App/assets/2319350/0650b846-c401-4aa9-8548-2fc61b19690a)

gedu commented:
@shawnborton updated, I overridden the common style because I was seeing too much space between the items, but I just remove it

<img width="401" alt="Screen Shot 2023-06-01 at 16 57 55" src="https://github.com/Expensify/App/assets/1676818/e81edc61-a529-4dc6-9ff0-d60c0f0f0fd3">


<img width="412" alt="Screen Shot 2023-06-01 at 16 58 01" src="https://github.com/Expensify/App/assets/1676818/fd1047ca-439e-4e69-9de8-be47d7f24507">

<img width="394" alt="Screen Shot 2023-06-01 at 16 58 23" src="https://github.com/Expensify/App/assets/1676818/36453739-1ee8-49e4-9ab2-1de8f031cc46">

All pushed

shawnborton commented:
The label for Country is too small when there is no value set - can you take a look at that? It should be the same size as our normal font, and then it becomes smaller once there is a value set.

gedu commented:
@shawnborton done
<img width="410" alt="Screen Shot 2023-06-01 at 21 05 33" src="https://github.com/Expensify/App/assets/1676818/09cb9a1d-c0f1-40cc-b08f-e4f30301a616">
<img width="398" alt="Screen Shot 2023-06-01 at 21 06 28" src="https://github.com/Expensify/App/assets/1676818/3707f127-c4fc-46a7-b882-abd7e64ac76f">

shawnborton commented:
Much better, thank you!

gedu commented:
@cristipaval should be good now

cristipaval commented:
@gedu Could you please resolve the conflicts?

gedu commented:
> @gedu Could you please resolve the conflicts?

Yes, working on it

cristipaval commented:
> > @gedu Could you please resolve the conflicts?
> 
> Yes, working on it

Thank you!

gedu commented:
@cristipaval @0xmiroslav Conflicts solved

gedu commented:
@0xmiroslav I'm having some issues trying to sign previous commits, I tried many commands also some shared into slack but no luck. can I create a new PR? what do you think?

gedu commented:
> @gedu we don't recommend force-push once review started but try only this time

@0xmiroslav  what do you mean by a force-push? That won't sign the old commits, there are 3 commits that need to be signed, I could sign one of the last ones but those that are from May are getting hard

gedu commented:
> Maybe you can push all of work so far in one commit. You're about to do this in new PR, right?

Yeah, I will create a new PR, because one of the unsigned commits is from May, and rebasing is really complicated

gedu commented:
@0xmiroslav pushed, sorry for the inconvenience, my main Mac is broken so using my backup Mac I forgot to sign the commits

venture1981 reviewed (commented):
seems like most of them have been fixed.

gedu commented:
@0xmiroslav any updates?

gedu commented:
@0xmiroslav 

- pre-populate search input and auto-scroll to selected item: FIXED
- BUG: weird transition animation (the entire page flickers): New Nav bug, not related with this happens in other screens
- BUG: after saving state, country is reverted: This happens in production, I would create another issue to solve this
- BUG: space between address line 1 and line 2 is bigger than others: FIXED
- BUG: after selecting address from auto-complete menu: FIXED

<img width="305" alt="Screen Shot 2023-06-19 at 11 48 00" src="https://github.com/Expensify/App/assets/1676818/7e9a0174-6ab1-4cd7-8e5a-ff5bbee163c4">


https://github.com/Expensify/App/assets/1676818/3cd25be0-6332-4009-bf44-c2910b805e65


Navigation bug in another screen

https://github.com/Expensify/App/assets/1676818/a6d5e4f3-2e9d-4063-858e-e5458ea9ef35
